xontrib/z.py

```diff
@@ -86,14 +86,14 @@ def load_data(self):
         with open(self.Z_DATA, 'rt') as f:
             for l in f:
                 l = l.strip()
-                p, r, t = l.split('|')
-                if '.' in r:
+                try:
+                    p, r, t = l.rsplit('|', 2)
                     r = float(r)
-                else:
-                    r = int(r)
-                if r >= 1:
-                    t = datetime.datetime.utcfromtimestamp(int(t))
-                    yield ZEntry(p, r, t)
+                    if r >= 1:
+                        t = datetime.datetime.utcfromtimestamp(float(t))
+                        yield ZEntry(p.replace('\\n','\n'), r, t)
+                except:
```

Comment:
Make `except Exception:`, so you don't catch `KeyboardInterrupt` and `SystemExit`